### ISSUE TITLE
fix finds 255 violations and can fix none of them

### ISSUE THREAD

**Issue by safareli** [sqlfluff/sqlfluff#4113]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

For big jinja sql file with 252 violations, sqlfluff fix fixes nothing

### Expected Behaviour

It should fix something

### Observed Behaviour

nothing is fixed

### How to reproduce

<details><summary>sql file</summary>
```sql

{%- set names = ['A','B','C', 'D', 'E', 'F', 'G','H', 'I', 'J', 'K'] -%}

with y as (

  select
    a,
		b,
		replace(c, ' ', '_') as c1,
    d,
    e,
    sum(f) as f,
		sum(g) as g,
    count(h) as hc
  from {{ref('my_first_dbt_model')}}
  {{group_by(5)}}
),

x as (

select
  a,
	b,
	c,
	{% for name in names %}
	sum(case when d = '{{name}}'
		and e = '1'
		then d else 0 end)  as {{name}}_e,
	{% endfor %}

	{% for name in names %}
	sum(case when d = '{{name}}'
		and e = '2'
		then d else 0 end)  as {{name}}_e,
	{% endfor %}

	{% for name in names %}
	sum(case when d = '{{name}}'
		and e = '3'
		then d else 0 end)  as {{name}}_e,
	{% endfor %}

	{% for name in names %}
	sum(case when d = '{{name}}'
		and e = '4'
				then d else 0 end)  as {{name}}_e,
	{% endfor %}

	{% for name in names %}
	sum(case when d = '{{name}}'
		then f else 0 end)  as {{name}}_f
	{% if not loop.last %},{% endif %}
	{% endfor %}

from y
where d in ('A','B','C', 'D', 'E', 'F', 'G','H', 'I', 'J', 'K')

{{group_by(3)}}
)

select
  *

from x

```
</details>

it's also very slow:
```
➜  sqlfluff_repro git:(master) ✗ time sqlfluff fix models/example/c.sql > fix.log
  [2 templating/parsing errors found]                                                                                             
sqlfluff fix models/example/c.sql > fix.log  16.69s user 0.14s system 99% cpu 16.872 total                                        
➜  sqlfluff_repro git:(master) ✗ time sqlfluff lint models/example/c.sql > lint.log
sqlfluff lint models/example/c.sql > lint.log  5.32s user 0.10s system 98% cpu 5.519 total  
```

<details><summary>fix.log</summary> 

```
==== finding fixable violations ====
== [models/example/c.sql] FAIL
L:   6 | P:   3 | L003 | Expected 1 indentation, found less than 1 [compared to
                       | line 01]
L:   8 | P:   3 | L003 | Expected 1 indentation, found 2 [compared to line 04]
L:   9 | P:   3 | L003 | Expected 1 indentation, found 2 [compared to line 04]
L:  13 | P:   3 | L003 | Expected 1 indentation, found 2 [compared to line 09]
L:  15 | P:   3 | L003 | Expected 1 indentation, found less than 1 [compared to
                       | line 01]
L:  15 | P:   8 | L046 | Jinja tags should have a single whitespace on either
                       | side: {{ref('my_first_dbt_model')}}
L:  16 | P:   1 | L001 | Unnecessary trailing whitespace.
L:  16 | P:   3 | L046 | Jinja tags should have a single whitespace on either
                       | side: {{group_by(5)}}
L:  21 | P:   1 | L003 | Expected 1 indentation, found 0 [compared to line 16]
L:  22 | P:   3 | L003 | Expected 2 indentations, found less than 1 [compared to
                       | line 16]
L:  23 | P:   1 | L004 | Incorrect indentation type found in file.
L:  24 | P:   1 | L004 | Incorrect indentation type found in file.
L:  25 | P:   1 | L004 | Incorrect indentation type found in file.
L:  26 | P:   1 | L004 | Incorrect indentation type found in file.
L:  26 | P:   1 | L004 | Incorrect indentation type found in file.
L:  26 | P:   1 | L004 | Incorrect indentation type found in file.
L:  26 | P:   1 | L004 | Incorrect indentation type found in file.
L:  26 | P:   1 | L004 | Incorrect indentation type found in file.
L:  26 | P:   1 | L004 | Incorrect indentation type found in file.
L:  26 | P:   1 | L004 | Incorrect indentation type found in file.
L:  26 | P:   1 | L004 | Incorrect indentation type found in file.
L:  26 | P:   1 | L004 | Incorrect indentation type found in file.
L:  26 | P:   1 | L004 | Incorrect indentation type found in file.
L:  26 | P:   1 | L004 | Incorrect indentation type found in file.
L:  26 | P:  20 | L046 | Jinja tags should have a single whitespace on either
                       | side: {{name}}
L:  27 | P:   1 | L004 | Incorrect indentation type found in file.
L:  27 | P:   1 | L004 | Incorrect indentation type found in file.
L:  27 | P:   1 | L004 | Incorrect indentation type found in file.
L:  27 | P:   1 | L004 | Incorrect indentation type found in file.
L:  27 | P:   1 | L004 | Incorrect indentation type found in file.
L:  27 | P:   1 | L004 | Incorrect indentation type found in file.
L:  27 | P:   1 | L004 | Incorrect indentation type found in file.
L:  27 | P:   1 | L004 | Incorrect indentation type found in file.
L:  27 | P:   1 | L004 | Incorrect indentation type found in file.
L:  27 | P:   1 | L004 | Incorrect indentation type found in file.
L:  27 | P:   1 | L004 | Incorrect indentation type found in file.
L:  28 | P:   1 | L004 | Incorrect indentation type found in file.
L:  28 | P:   1 | L004 | Incorrect indentation type found in file.
L:  28 | P:   1 | L004 | Incorrect indentation type found in file.
L:  28 | P:   1 | L004 | Incorrect indentation type found in file.
L:  28 | P:   1 | L004 | Incorrect indentation type found in file.
L:  28 | P:   1 | L004 | Incorrect indentation type found in file.
L:  28 | P:   1 | L004 | Incorrect indentation type found in file.
L:  28 | P:   1 | L004 | Incorrect indentation type found in file.
L:  28 | P:   1 | L004 | Incorrect indentation type found in file.
L:  28 | P:   1 | L004 | Incorrect indentation type found in file.
L:  28 | P:   1 | L004 | Incorrect indentation type found in file.
L:  28 | P:  26 | L046 | Jinja tags should have a single whitespace on either
                       | side: {{name}}
L:  29 | P:   1 | L004 | Incorrect indentation type found in file.
L:  29 | P:   1 | L004 | Incorrect indentation type found in file.
L:  29 | P:   1 | L004 | Incorrect indentation type found in file.
L:  29 | P:   1 | L004 | Incorrect indentation type found in file.
L:  29 | P:   1 | L004 | Incorrect indentation type found in file.
L:  29 | P:   1 | L004 | Incorrect indentation type found in file.
L:  29 | P:   1 | L004 | Incorrect indentation type found in file.
L:  29 | P:   1 | L004 | Incorrect indentation type found in file.
L:  29 | P:   1 | L004 | Incorrect indentation type found in file.
L:  29 | P:   1 | L004 | Incorrect indentation type found in file.
L:  29 | P:   1 | L004 | Incorrect indentation type found in file.
L:  31 | P:   1 | L004 | Incorrect indentation type found in file.
L:  32 | P:   1 | L004 | Incorrect indentation type found in file.
L:  32 | P:   1 | L004 | Incorrect indentation type found in file.
L:  32 | P:   1 | L004 | Incorrect indentation type found in file.
L:  32 | P:   1 | L004 | Incorrect indentation type found in file.
L:  32 | P:   1 | L004 | Incorrect indentation type found in file.
L:  32 | P:   1 | L004 | Incorrect indentation type found in file.
L:  32 | P:   1 | L004 | Incorrect indentation type found in file.
L:  32 | P:   1 | L004 | Incorrect indentation type found in file.
L:  32 | P:   1 | L004 | Incorrect indentation type found in file.
L:  32 | P:   1 | L004 | Incorrect indentation type found in file.
L:  32 | P:   1 | L004 | Incorrect indentation type found in file.
L:  32 | P:  20 | L046 | Jinja tags should have a single whitespace on either
                       | side: {{name}}
L:  33 | P:   1 | L004 | Incorrect indentation type found in file.
L:  33 | P:   1 | L004 | Incorrect indentation type found in file.
L:  33 | P:   1 | L004 | Incorrect indentation type found in file.
L:  33 | P:   1 | L004 | Incorrect indentation type found in file.
L:  33 | P:   1 | L004 | Incorrect indentation type found in file.
L:  33 | P:   1 | L004 | Incorrect indentation type found in file.
L:  33 | P:   1 | L004 | Incorrect indentation type found in file.
L:  33 | P:   1 | L004 | Incorrect indentation type found in file.
L:  33 | P:   1 | L004 | Incorrect indentation type found in file.
L:  33 | P:   1 | L004 | Incorrect indentation type found in file.
L:  33 | P:   1 | L004 | Incorrect indentation type found in file.
L:  34 | P:   1 | L004 | Incorrect indentation type found in file.
L:  34 | P:   1 | L004 | Incorrect indentation type found in file.
L:  34 | P:   1 | L004 | Incorrect indentation type found in file.
L:  34 | P:   1 | L004 | Incorrect indentation type found in file.
L:  34 | P:   1 | L004 | Incorrect indentation type found in file.
L:  34 | P:   1 | L004 | Incorrect indentation type found in file.
L:  34 | P:   1 | L004 | Incorrect indentation type found in file.
L:  34 | P:   1 | L004 | Incorrect indentation type found in file.
L:  34 | P:   1 | L004 | Incorrect indentation type found in file.
L:  34 | P:   1 | L004 | Incorrect indentation type found in file.
L:  34 | P:   1 | L004 | Incorrect indentation type found in file.
L:  34 | P:  26 | L046 | Jinja tags should have a single whitespace on either
                       | side: {{name}}
L:  35 | P:   1 | L004 | Incorrect indentation type found in file.
L:  35 | P:   1 | L004 | Incorrect indentation type found in file.
L:  35 | P:   1 | L004 | Incorrect indentation type found in file.
L:  35 | P:   1 | L004 | Incorrect indentation type found in file.
L:  35 | P:   1 | L004 | Incorrect indentation type found in file.
L:  35 | P:   1 | L004 | Incorrect indentation type found in file.
L:  35 | P:   1 | L004 | Incorrect indentation type found in file.
L:  35 | P:   1 | L004 | Incorrect indentation type found in file.
L:  35 | P:   1 | L004 | Incorrect indentation type found in file.
L:  35 | P:   1 | L004 | Incorrect indentation type found in file.
L:  35 | P:   1 | L004 | Incorrect indentation type found in file.
L:  37 | P:   1 | L004 | Incorrect indentation type found in file.
L:  38 | P:   1 | L004 | Incorrect indentation type found in file.
L:  38 | P:   1 | L004 | Incorrect indentation type found in file.
L:  38 | P:   1 | L004 | Incorrect indentation type found in file.
L:  38 | P:   1 | L004 | Incorrect indentation type found in file.
L:  38 | P:   1 | L004 | Incorrect indentation type found in file.
L:  38 | P:   1 | L004 | Incorrect indentation type found in file.
L:  38 | P:   1 | L004 | Incorrect indentation type found in file.
L:  38 | P:   1 | L004 | Incorrect indentation type found in file.
L:  38 | P:   1 | L004 | Incorrect indentation type found in file.
L:  38 | P:   1 | L004 | Incorrect indentation type found in file.
L:  38 | P:   1 | L004 | Incorrect indentation type found in file.
L:  38 | P:  20 | L046 | Jinja tags should have a single whitespace on either
                       | side: {{name}}
L:  39 | P:   1 | L004 | Incorrect indentation type found in file.
L:  39 | P:   1 | L004 | Incorrect indentation type found in file.
L:  39 | P:   1 | L004 | Incorrect indentation type found in file.
L:  39 | P:   1 | L004 | Incorrect indentation type found in file.
L:  39 | P:   1 | L004 | Incorrect indentation type found in file.
L:  39 | P:   1 | L004 | Incorrect indentation type found in file.
L:  39 | P:   1 | L004 | Incorrect indentation type found in file.
L:  39 | P:   1 | L004 | Incorrect indentation type found in file.
L:  39 | P:   1 | L004 | Incorrect indentation type found in file.
L:  39 | P:   1 | L004 | Incorrect indentation type found in file.
L:  39 | P:   1 | L004 | Incorrect indentation type found in file.
L:  40 | P:   1 | L004 | Incorrect indentation type found in file.
L:  40 | P:   1 | L004 | Incorrect indentation type found in file.
L:  40 | P:   1 | L004 | Incorrect indentation type found in file.
L:  40 | P:   1 | L004 | Incorrect indentation type found in file.
L:  40 | P:   1 | L004 | Incorrect indentation type found in file.
L:  40 | P:   1 | L004 | Incorrect indentation type found in file.
L:  40 | P:   1 | L004 | Incorrect indentation type found in file.
L:  40 | P:   1 | L004 | Incorrect indentation type found in file.
L:  40 | P:   1 | L004 | Incorrect indentation type found in file.
L:  40 | P:   1 | L004 | Incorrect indentation type found in file.
L:  40 | P:   1 | L004 | Incorrect indentation type found in file.
L:  40 | P:  26 | L046 | Jinja tags should have a single whitespace on either
                       | side: {{name}}
L:  41 | P:   1 | L004 | Incorrect indentation type found in file.
L:  41 | P:   1 | L004 | Incorrect indentation type found in file.
L:  41 | P:   1 | L004 | Incorrect indentation type found in file.
L:  41 | P:   1 | L004 | Incorrect indentation type found in file.
L:  41 | P:   1 | L004 | Incorrect indentation type found in file.
L:  41 | P:   1 | L004 | Incorrect indentation type found in file.
L:  41 | P:   1 | L004 | Incorrect indentation type found in file.
L:  41 | P:   1 | L004 | Incorrect indentation type found in file.
L:  41 | P:   1 | L004 | Incorrect indentation type found in file.
L:  41 | P:   1 | L004 | Incorrect indentation type found in file.
L:  41 | P:   1 | L004 | Incorrect indentation type found in file.
L:  43 | P:   1 | L004 | Incorrect indentation type found in file.
L:  44 | P:   1 | L004 | Incorrect indentation type found in file.
L:  44 | P:   1 | L004 | Incorrect indentation type found in file.
L:  44 | P:   1 | L004 | Incorrect indentation type found in file.
L:  44 | P:   1 | L004 | Incorrect indentation type found in file.
L:  44 | P:   1 | L004 | Incorrect indentation type found in file.
L:  44 | P:   1 | L004 | Incorrect indentation type found in file.
L:  44 | P:   1 | L004 | Incorrect indentation type found in file.
L:  44 | P:   1 | L004 | Incorrect indentation type found in file.
L:  44 | P:   1 | L004 | Incorrect indentation type found in file.
L:  44 | P:   1 | L004 | Incorrect indentation type found in file.
L:  44 | P:   1 | L004 | Incorrect indentation type found in file.
L:  44 | P:  20 | L046 | Jinja tags should have a single whitespace on either
                       | side: {{name}}
L:  45 | P:   1 | L004 | Incorrect indentation type found in file.
L:  45 | P:   1 | L004 | Incorrect indentation type found in file.
L:  45 | P:   1 | L004 | Incorrect indentation type found in file.
L:  45 | P:   1 | L004 | Incorrect indentation type found in file.
L:  45 | P:   1 | L004 | Incorrect indentation type found in file.
L:  45 | P:   1 | L004 | Incorrect indentation type found in file.
L:  45 | P:   1 | L004 | Incorrect indentation type found in file.
L:  45 | P:   1 | L004 | Incorrect indentation type found in file.
L:  45 | P:   1 | L004 | Incorrect indentation type found in file.
L:  45 | P:   1 | L004 | Incorrect indentation type found in file.
L:  45 | P:   1 | L004 | Incorrect indentation type found in file.
L:  46 | P:   5 | L003 | Expected 2 indentations, found 4 [compared to line 162]
L:  46 | P:   5 | L003 | Expected 2 indentations, found 4 [compared to line 166]
L:  46 | P:   5 | L003 | Expected 2 indentations, found 4 [compared to line 170]
L:  46 | P:   5 | L003 | Expected 2 indentations, found 4 [compared to line 174]
L:  46 | P:   5 | L003 | Expected 2 indentations, found 4 [compared to line 178]
L:  46 | P:   5 | L003 | Expected 2 indentations, found 4 [compared to line 182]
L:  46 | P:   5 | L003 | Expected 2 indentations, found 4 [compared to line 186]
L:  46 | P:   5 | L003 | Expected 2 indentations, found 4 [compared to line 190]
L:  46 | P:   5 | L003 | Expected 2 indentations, found 4 [compared to line 194]
L:  46 | P:   5 | L003 | Expected 2 indentations, found 4 [compared to line 198]
L:  46 | P:   5 | L003 | Expected 2 indentations, found 4 [compared to line 202]
L:  46 | P:  28 | L046 | Jinja tags should have a single whitespace on either
                       | side: {{name}}
L:  47 | P:   1 | L004 | Incorrect indentation type found in file.
L:  47 | P:   1 | L004 | Incorrect indentation type found in file.
L:  47 | P:   1 | L004 | Incorrect indentation type found in file.
L:  47 | P:   1 | L004 | Incorrect indentation type found in file.
L:  47 | P:   1 | L004 | Incorrect indentation type found in file.
L:  47 | P:   1 | L004 | Incorrect indentation type found in file.
L:  47 | P:   1 | L004 | Incorrect indentation type found in file.
L:  47 | P:   1 | L004 | Incorrect indentation type found in file.
L:  47 | P:   1 | L004 | Incorrect indentation type found in file.
L:  47 | P:   1 | L004 | Incorrect indentation type found in file.
L:  47 | P:   1 | L004 | Incorrect indentation type found in file.
L:  49 | P:   1 | L004 | Incorrect indentation type found in file.
L:  50 | P:   1 | L004 | Incorrect indentation type found in file.
L:  50 | P:   1 | L004 | Incorrect indentation type found in file.
L:  50 | P:   1 | L004 | Incorrect indentation type found in file.
L:  50 | P:   1 | L004 | Incorrect indentation type found in file.
L:  50 | P:   1 | L004 | Incorrect indentation type found in file.
L:  50 | P:   1 | L004 | Incorrect indentation type found in file.
L:  50 | P:   1 | L004 | Incorrect indentation type found in file.
L:  50 | P:   1 | L004 | Incorrect indentation type found in file.
L:  50 | P:   1 | L004 | Incorrect indentation type found in file.
L:  50 | P:   1 | L004 | Incorrect indentation type found in file.
L:  50 | P:   1 | L004 | Incorrect indentation type found in file.
L:  50 | P:  20 | L046 | Jinja tags should have a single whitespace on either
                       | side: {{name}}
L:  51 | P:   1 | L004 | Incorrect indentation type found in file.
L:  51 | P:   1 | L004 | Incorrect indentation type found in file.
L:  51 | P:   1 | L004 | Incorrect indentation type found in file.
L:  51 | P:   1 | L004 | Incorrect indentation type found in file.
L:  51 | P:   1 | L004 | Incorrect indentation type found in file.
L:  51 | P:   1 | L004 | Incorrect indentation type found in file.
L:  51 | P:   1 | L004 | Incorrect indentation type found in file.
L:  51 | P:   1 | L004 | Incorrect indentation type found in file.
L:  51 | P:   1 | L004 | Incorrect indentation type found in file.
L:  51 | P:   1 | L004 | Incorrect indentation type found in file.
L:  51 | P:   1 | L004 | Incorrect indentation type found in file.
L:  51 | P:  26 | L046 | Jinja tags should have a single whitespace on either
                       | side: {{name}}
L:  52 | P:   1 | L004 | Incorrect indentation type found in file.
L:  52 | P:   1 | L004 | Incorrect indentation type found in file.
L:  52 | P:   1 | L004 | Incorrect indentation type found in file.
L:  52 | P:   1 | L004 | Incorrect indentation type found in file.
L:  52 | P:   1 | L004 | Incorrect indentation type found in file.
L:  52 | P:   1 | L004 | Incorrect indentation type found in file.
L:  52 | P:   1 | L004 | Incorrect indentation type found in file.
L:  52 | P:   1 | L004 | Incorrect indentation type found in file.
L:  52 | P:   1 | L004 | Incorrect indentation type found in file.
L:  52 | P:   1 | L004 | Incorrect indentation type found in file.
L:  52 | P:   1 | L004 | Incorrect indentation type found in file.
L:  53 | P:   1 | L004 | Incorrect indentation type found in file.
L:  53 | P:   1 | L004 | Incorrect indentation type found in file.
L:  53 | P:   1 | L004 | Incorrect indentation type found in file.
L:  53 | P:   1 | L004 | Incorrect indentation type found in file.
L:  53 | P:   1 | L004 | Incorrect indentation type found in file.
L:  53 | P:   1 | L004 | Incorrect indentation type found in file.
L:  53 | P:   1 | L004 | Incorrect indentation type found in file.
L:  53 | P:   1 | L004 | Incorrect indentation type found in file.
L:  53 | P:   1 | L004 | Incorrect indentation type found in file.
L:  53 | P:   1 | L004 | Incorrect indentation type found in file.
L:  53 | P:   1 | L004 | Incorrect indentation type found in file.
L:  55 | P:   1 | L003 | Expected 1 indentation, found 0 [compared to line 16]
L:  56 | P:   1 | L003 | Expected 1 indentation, found 0 [compared to line 16]
L:  56 | P:  17 | L048 | Missing whitespace before 'B'
L:  56 | P:  21 | L048 | Missing whitespace before 'C'
L:  56 | P:  45 | L048 | Missing whitespace before 'H'
L:  58 | P:   1 | L046 | Jinja tags should have a single whitespace on either
                       | side: {{group_by(3)}}
L:  61 | P:   1 | L036 | Select targets should be on a new line unless there is
                       | only one select target.
L:  62 | P:   3 | L003 | Expected 1 indentation, found less than 1 [compared to
                       | line 258]
==== no fixable linting violations found ====
All Finished!
  [255 unfixable linting violations found]

```
 </details>

### Dialect

snowflake

### Version

1.4.1

### Configuration

<details><summary>.sqlfluff</summary>

```
[sqlfluff]
dialect = snowflake
templater = jinja
nocolor = true
ignore_templated_areas = true
exclude_rules = L005,L007,L016,L008,L009,L031,L044,L042,L043,L045,L039,L025,L019,L059,L034

[sqlfluff:indentation]
indented_joins = false

[sqlfluff:templater]
unwrap_wrapped_queries = true

[sqlfluff:templater:jinja]
apply_dbt_builtins = true
load_macros_from_path = macros

[sqlfluff:templater:jinja:macros]
dbt_ref = {% macro ref(model_ref) %}{{model_ref}}{% endmacro %}
dbt_source = {% macro source(source_name, table) %}{{source_name}}_{{table}}{% endmacro %}
dbt_config = {% macro config() %}{% for k in kwargs %}{% endfor %}{% endmacro %}
dbt_var = {% macro var(variable, default='') %}item{% endmacro %}
dbt_is_incremental = {% macro is_incremental() %}True{% endmacro %}

[sqlfluff:rules]
max_line_length = 180
indent_unit = space
tab_space_size = 4

[sqlfluff:rules:L010]
capitalisation_policy = lower

[sqlfluff:rules:L014]
extended_capitalisation_policy = lower

[sqlfluff:rules:L030]
capitalisation_policy = lower

[sqlfluff:rules:L040]
capitalisation_policy = lower

[sqlfluff:rules:L054]
group_by_and_order_by_style = implicit

```

</details>

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by tunetheweb** [comment]
Raises #4114 as an idea to address the slowness, while we continue to track [longer term performance issues](https://github.com/sqlfluff/sqlfluff/issues?q=is%3Aopen+is%3Aissue+label%3Aperformance).
